core/src/main/scala/akka/persistence/r2dbc/query/scaladsl/R2dbcReadJournal.scala

```diff
@@ -112,7 +118,7 @@ final class R2dbcReadJournal(system: ExtendedActorSystem, config: Config, cfgPat
         metadata,
         row.entityType,
         row.slice,
-        filtered = false,
+        filtered = event.isEmpty,
```

Comment:
This is probably not enough/correct because missing payload could also mean lazy loaded for backtracking

Reply:
Added a flag for lazy load in [a359c39](https://github.com/akka/akka-persistence-r2dbc/pull/416/commits/a359c390a6ccb56785c8a4258d3220cee158f857)